azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -284,7 +293,39 @@ public void cleanUpDanglingYarnApplications() {
     logger.info("Get the set of all executions: " +
         toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
 
+    if (executionStoppedFlows.isEmpty()) {
+      logger.info("No execution needs to kill yarn application, exit");
+      return;
+    }
+
     // TODO: For each of yarn clusters: get applications and kill those in the above set
+    for (Entry<String, Cluster> entry : this.allClusters.entrySet()) {
+      logger.info("clean up yarn applications in cluster:" + entry.getValue().clusterId);
+      cleanUpYarnApplicationsInCluster(toBeCleanedContainers, entry.getValue());
+    }
+  }
+
+  private void cleanUpYarnApplicationsInCluster(Set<Integer> toBeCleanedContainers,
```

Comment:
I think we need to make this whole class in the async path and gate it by feature flag

Reply:
Async logic is in the following PRs

Reply:
will this impact CleanupManager if you don't put it in async? or at least we can use feature flag to gate it.

Reply:
I mean if you merge this PR

Reply:
this method is not the one we do `schedule` on, the `cleanUpDanglingYarnApplications` will be scheduled asynchronously at a cadence

Reply:
and at this moment (this PR) these logics are not invoked at all, will start calling the logic at the final PR